mutiny-core/src/lib.rs

```diff
@@ -1280,7 +1275,54 @@ impl<S: MutinyStorage> MutinyWallet<S> {
             );
         }
 
-        Ok(final_result)
+        Ok(FedimintSweepResult {
+            amount: first_invoice_amount,
+            fees: Some(first_invoice_res.fees_paid.unwrap_or(0) + fee),
+        })
+    }
+
+    pub async fn estimate_sweep_federation_fee(
+        &self,
+        amount: Option<u64>,
+    ) -> Result<Option<u64>, MutinyError> {
+        if let Some(0) = amount {
+            return Ok(None);
+        }
+
+        // TODO support more than one federation
+        let federation_ids = self.list_federation_ids().await?;
+        if federation_ids.is_empty() {
+            return Err(MutinyError::NotFound);
+        }
+
+        let federation_id = &federation_ids[0];
+        let federation_lock = self.federations.read().await;
+        let fedimint_client = federation_lock
+            .get(federation_id)
+            .ok_or(MutinyError::NotFound)?;
+
+        // if the user provided amount, this is easy
+        if let Some(amt) = amount {
+            let (_inv, fee) = self.node_manager.create_invoice(amt).await?;
```

Comment:
if we do it this way then we'll fill up the storage with unused invoices. Can't we just call `get_lsp_fee_msat` instead